### PR TITLE
Add getRelevantErrata endpoint that takes a list of sids

### PR DESCRIPTION
**Pull request by dottorblaster**
## What does this PR change?

Solves (at least I hope so) #8183.

Adds a new endpoint to the API that lets `getRelevantErrata` have a list of `sid`s as a parameter. 

## GUI diff

No difference.

Before:

After:

- [X] **DONE**

## Documentation
- API documentation added: please review the Wiki page [Writing Documentation for the API](https://github.com/uyuni-project/uyuni/wiki/Writing-documentation-for-the-API) if you have any changes to API documentation.
- I hope I got it right!

- [X] **DONE**

## Test coverage
- Unit tests were added

- [X] **DONE**

## Links

Issue(s): #8183 

- [X] **DONE**

## Changelogs

I still need to write an entry.

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Review by rjmateus — APPROVED** [review]
Thank you for the contribution. Looks good to me but lets wait for Can review also.

**Comment by dottorblaster** [comment]
Thank you so much @rjmateus! I'm fixing the changelog as well as soon as this build finished with satisfying results on my end :smile:

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java [hunk 1]
```diff
@@ -3485,6 +3485,45 @@ public List<ErrataOverview> getRelevantErrata(User loggedInUser, Integer sid) {
         return dr;
     }
 
+    /**
+     * Returns a list of all errata that are relevant to the system.
```
Comment (cbbayburt) [inline java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java:3489]:
```suggestion
     * Returns a list of all errata that are relevant to a list of systems.
```

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java [hunk 2]
```diff
@@ -3485,6 +3485,45 @@ public List<ErrataOverview> getRelevantErrata(User loggedInUser, Integer sid) {
...
+     *
+     * @param loggedInUser The current user
+     * @param sids The ids of the systems in question
+     * @return Returns an array of maps representing the errata that can be applied
+     * @throws FaultException A FaultException is thrown if the server corresponding to
+     * sid cannot be found.
+     *
+     * @apidoc.doc Returns a list of all errata that are relevant to the system.
```
Comment (cbbayburt) [inline java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java:3497]:
```suggestion
     * @apidoc.doc Returns a list of all errata that are relevant to a list of systems.
```

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java [hunk 3]
```diff
@@ -3485,6 +3485,45 @@ public List<ErrataOverview> getRelevantErrata(User loggedInUser, Integer sid) {
...
     }
 
+    /**
+     * Returns a list of all errata that are relevant to the system.
+     *
+     * @param loggedInUser The current user
+     * @param sids The ids of the systems in question
+     * @return Returns an array of maps representing the errata that can be applied
+     * @throws FaultException A FaultException is thrown if the server corresponding to
+     * sid cannot be found.
```
Comment (cbbayburt) [inline java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java:3495]:
```suggestion
     * @throws FaultException A FaultException is thrown if any server in the list cannot be found.
```

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java [hunk 4]
```diff
@@ -3485,6 +3485,45 @@ public List<ErrataOverview> getRelevantErrata(User loggedInUser, Integer sid) {
         return dr;
     }
 
+    /**
+     * Returns a list of all errata that are relevant to the system.
+     *
+     * @param loggedInUser The current user
+     * @param sids The ids of the systems in question
+     * @return Returns an array of maps representing the errata that can be applied
```
Comment (cbbayburt) [inline java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java:3493]:
```suggestion
     * @return Returns an array of maps representing the errata that can be applied to each system
```

**Review by cbbayburt — CHANGES_REQUESTED** [review]
Thanks for the contribution, @dottorblaster.

I just requested some minor updates for the documentation to reflect the change. FYI, that the actual API documentation is generated from the javadoc entries that start with `@apidoc.`

**Review by cbbayburt — APPROVED** [review]

**Pull request merged**